### DISCUSSION TITLE
dpgen simplify for transfer learning

### DISCUSSION THREAD

**Discussion by alikhamze** [deepmodeling/dpgen#1417] [Q&A]
Hello,

I am trying to use the simplify function of dpgen for transfer learning, as [this document](https://tutorials.deepmodeling.com/en/latest/CaseStudies/Transfer-learning/Transfer-learning.html#when-to-use-transfer-learning) says is possible.

I set up my simplify.json following the examples [here](https://github.com/deepmodeling/dpgen/blob/master/examples/simplify-MAPbI3-scan-lebesgue/simplify_example/simplify.json) and [here](https://tutorials.deepmodeling.com/en/latest/CaseStudies/Transfer-learning/Transfer-learning.html#example-ag-au), with the change that for the new, transfer-learned model, I changed the type map to replace the old element with the new element.

When I run simplify, I get the following error: `AssertionError: size of fp_pp_files should be the same as the size of type_map` from line 2989 in `run.py`. 
This is because simplify overrides the type map in the parameter file and adds the old species to the type map, and then crashes because I have not added the old species' POTCAR to the simplify input file.

How can I set up my parameter file for transfer learning? Do I need to add the old POTCAR to the parameter file? I think doing so would fix the error, but it's not clear to me that the transfer learning would proceed as I want it to if I do...

Thank you!
-Ali

**Comment by njzjz** [comment]
There are two `type_map` parameters in the file. Did you only change one?

**Reply by alikhamze** [reply]
@njzjz both have been updated to the desired type map.

**Reply by alikhamze** [reply]
@njzjz i've attached my input file, do you see any other possible errors? Thank you.
[simplify.json](https://github.com/deepmodeling/dpgen/files/13604325/simplify.json)

**Comment by h757062840** [comment]
May I ask if the problem has been resolved? I have also encountered the same problem

**Reply by njzjz** [reply]
It will be helpful if you could print those two values.
